modules/core/src/test/java/org/apache/ignite/testframework/junits/GridAbstractTest.java

```diff
@@ -748,6 +752,54 @@ private void cleanUpTestEnviroment() throws Exception {
         }
     }
 
+    /**
+     * Runs afterTest() callback method with necessary scaffolding.
+     */
+    private void runAfterTest() throws Exception {
+        AtomicBoolean afterTestFinished = new AtomicBoolean(false);
+
+        ScheduledExecutorService scheduler = scheduleThreadDumpOnAfterTestTimeOut(afterTestFinished);
+
+        try {
+            afterTest();
```

Comment:
If this hangs forever, we won't be shutting down scheduler. But it should be ok, right?